<!-- -->
<launch>
	<arg name="maxVelocity" 				default="3.0" />
	<arg name="minVelocity" 				default="0.0" />
	<arg name="velocityProfileFactor" 		default="1.0" />
	<arg name="maxPlanningDistance" 		default="1000000.0" />
	<arg name="maxLocalPlanDistance" 		default="40.0" />
	<arg name="samplingTipMargin" 			default="2.0"  />
	<arg name="samplingOutMargin" 			default="10.0" />
	<arg name="samplingSpeedFactor" 		default="0.25" />
	<arg name="pathDensity" 				default="0.25" />
	<arg name="rollOutDensity" 				default="0.5" />
	<arg name="rollOutsNumber" 				default="4"    />
	<arg name="horizonDistance" 			default="200"  />
	
	<arg name="minFollowingDistance" 		default="16.0"  />
	<arg name="maxFollowingDistance" 		default="1000.0" /> <!-- Expired -->
	<arg name="minDistanceToAvoid" 			default="16.0" />
	<arg name="maxDistanceToAvoid" 			default="6.0"  />
	<arg name="speedProfileFactor"			default="1.0"  />
	
	<arg name="horizontalSafetyDistancel"	default="0.3"  />
	<arg name="verticalSafetyDistance"		default="0.7"  />
	
	<arg name="enableSwerving" 				default="true"  />
	<arg name="enableFollowing" 			default="true" />
	<arg name="enableHeadingSmoothing" 		default="false" />
	<arg name="enableTrafficLightBehavior" 	default="true" />
	<arg name="enableStopSignBehavior" 		default="true" />	
	<arg name="enableLaneChange" 			default="false" />
	
	<arg name="width" 						default="1.85"  />
	<arg name="length" 						default="4.2"  />
	<arg name="wheelBaseLength" 			default="2.7"  />
	<arg name="turningRadius"				default="5.2"  />
	<arg name="maxSteerAngle" 				default="0.5" />
	
	<arg name="enableObjectTracking" 		default="true" /> <!-- use internal kalman filter for track obstacles " -->
	<arg name="enableOutsideControl" 		default="false" /> <!-- use internal kalman filter for track obstacles " -->	
	<arg name="enableOdometryStatus"		default="false" /> <!-- read velocities from Odometry from robot "" -->
	
	<arg name="mapSource" 					default="1" /> <!-- Autoware=0, Vector Map Folder=1, kml=2 -->
	<arg name="mapFileName" 				default="/media/hatem/8ac0c5d5-8793-4b98-8728-55f8d67ec0f4/data/ToyotaCity2/map/vector_map/" />

	<node pkg="dp_planner" type="dp_planner" name="dp_planner" output="screen">
		<param name="maxVelocity" 					value="$(arg maxVelocity)" />
	    <param name="minVelocity" 					value="$(arg minVelocity)" />
	    
	    <param name="velocityProfileFactor" 		value="$(arg velocityProfileFactor)" />
		<param name="maxPlanningDistance" 			value="$(arg maxPlanningDistance)" />
		<param name="maxLocalPlanDistance" 			value="$(arg maxLocalPlanDistance)" />
		<param name="samplingTipMargin" 			value="$(arg samplingTipMargin)" />
		<param name="samplingOutMargin" 			value="$(arg samplingOutMargin)" />
		<param name="samplingSpeedFactor" 			value="$(arg samplingSpeedFactor)" />
		<param name="pathDensity" 					value="$(arg pathDensity)" />
		<param name="rollOutDensity" 				value="$(arg rollOutDensity)" />
		<param name="rollOutsNumber" 				value="$(arg rollOutsNumber)" />
		<param name="horizonDistance" 				value="$(arg horizonDistance)" />
		
		<param name="minFollowingDistance" 			value="$(arg minFollowingDistance)" />
		<param name="maxFollowingDistance" 			value="$(arg maxFollowingDistance)" />
		<param name="minDistanceToAvoid" 			value="$(arg minDistanceToAvoid)" />
		<param name="maxDistanceToAvoid" 			value="$(arg maxDistanceToAvoid)" />
		<param name="speedProfileFactor"			value="$(arg speedProfileFactor)" />
		
		<param name="horizontalSafetyDistancel"		value="$(arg horizontalSafetyDistancel)" />
		<param name="verticalSafetyDistance"		value="$(arg verticalSafetyDistance)" />
		
		<param name="enableSwerving" 				value="$(arg enableSwerving)" />
		<param name="enableFollowing" 				value="$(arg enableFollowing)" />
		<param name="enableHeadingSmoothing" 		value="$(arg enableHeadingSmoothing)" />
		<param name="enableTrafficLightBehavior" 	value="$(arg enableTrafficLightBehavior)" />
		<param name="enableStopSignBehavior" 		value="$(arg enableStopSignBehavior)" />		
		<param name="enableLaneChange" 				value="$(arg enableLaneChange)" />
		
		<param name="width" 						value="$(arg width)" />
		<param name="length" 						value="$(arg length)" />
		<param name="wheelBaseLength" 				value="$(arg wheelBaseLength)" />
		<param name="turningRadius" 				value="$(arg turningRadius)" />
		<param name="maxSteerAngle" 				value="$(arg maxSteerAngle)" />
		
		<param name="enableObjectTracking" 			value="$(arg enableObjectTracking)" />
		<param name="enableOutsideControl" 			value="$(arg enableOutsideControl)" />
		
		<param name="enableOdometryStatus" 			value="$(arg enableOdometryStatus)" />
		<param name="mapSource" 					value="$(arg mapSource)" />
		<param name="mapFileName" 					value="$(arg mapFileName)" />

	</node>

</launch>